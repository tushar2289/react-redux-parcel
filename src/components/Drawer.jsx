import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { useSelector, shallowEqual, useDispatch } from 'react-redux';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import Link from '@material-ui/core/Link';
import Badge from '@material-ui/core/Badge';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import Switch from '@material-ui/core/Switch';
import Tooltip from '@material-ui/core/Tooltip';
import Brightness5Icon from '@material-ui/icons/Brightness5';
import Brightness3Icon from '@material-ui/icons/Brightness3';
import AccountBalanceIcon from '@material-ui/icons/AccountBalance';
import RestaurantIcon from '@material-ui/icons/Restaurant';
import LocalHospitalIcon from '@material-ui/icons/LocalHospital';
import TrainIcon from '@material-ui/icons/Train';
import LocalAirportIcon from '@material-ui/icons/LocalAirport';
import LocalGasStationIcon from '@material-ui/icons/LocalGasStation';
import LocalGroceryStoreIcon from '@material-ui/icons/LocalGroceryStore';
import LocalAtmIcon from '@material-ui/icons/LocalAtm';
import LocalMallIcon from '@material-ui/icons/LocalMall';
import FavoriteIcon from '@material-ui/icons/Favorite';
import GitHubIcon from '@material-ui/icons/GitHub';
import FormLabel from '@material-ui/core/FormLabel';
import FormControl from '@material-ui/core/FormControl';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { Typography } from '@material-ui/core';
import {
  enablePlaceType,
  getPlaceTypes,
} from '../features/filters/placeTypeSlice';
import {
  RESTAURANT,
  BANK,
  HOSPITAL,
  ATM,
  AIRPORT,
  SUPERMARKET,
  TRAIN_STATION,
  GAS_STATION,
  SHOPPING_MALL,
} from '../constants/mapPlaceTypes';
import { setTheme, getTheme } from '../features/themeselector/slice';
import { getPlaces } from '../features/places/slice';

export const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  legend: {
    marginLeft: theme.spacing(2),
    color: theme.palette.text.primary,
  },
}));

const getIcon = (type) => {
  switch (type) {
    case RESTAURANT:
      return <RestaurantIcon />;
    case BANK:
      return <AccountBalanceIcon />;
    case HOSPITAL:
      return <LocalHospitalIcon />;
    case ATM:
      return <LocalAtmIcon />;
    case AIRPORT:
      return <LocalAirportIcon />;
    case SUPERMARKET:
      return <LocalGroceryStoreIcon />;
    case TRAIN_STATION:
      return <TrainIcon />;
    case GAS_STATION:
      return <LocalGasStationIcon />;
    case SHOPPING_MALL:
      return <LocalMallIcon />;
  }
};

const Drawer = () => {
  const classes = useStyles();
  const dispatch = useDispatch();

  const { theme } = useSelector(getTheme, shallowEqual);

  const toggleTheme = function () {
    dispatch(setTheme(event.target.checked ? 'dark' : 'light'));
  };

  const badgeCount = {
    [RESTAURANT]: 0,
    [BANK]: 0,
    [HOSPITAL]: 0,
    [ATM]: 0,
    [AIRPORT]: 0,
    [TRAIN_STATION]: 0,
    [SUPERMARKET]: 0,
    [GAS_STATION]: 0,
    [SHOPPING_MALL]: 0,
  };

  const { places } = useSelector(getPlaces, shallowEqual);

  const placeTypes = useSelector(getPlaceTypes, shallowEqual);

  const handleChange = (event) => {
    dispatch(
      enablePlaceType({ type: event.target.name, value: event.target.checked })
    );
  };

  places.forEach((place) => {
    if (place.types.indexOf(RESTAURANT) != -1) badgeCount[RESTAURANT]++;
    else if (place.types.indexOf(BANK) != -1) badgeCount[BANK]++;
    else if (place.types.indexOf(HOSPITAL) != -1) badgeCount[HOSPITAL]++;
    else if (place.types.indexOf(ATM) != -1) badgeCount[ATM]++;
    else if (place.types.indexOf(AIRPORT) != -1) badgeCount[AIRPORT]++;
    else if (place.types.indexOf(TRAIN_STATION) != -1)
      badgeCount[TRAIN_STATION]++;
    else if (place.types.indexOf(SUPERMARKET) != -1) badgeCount[SUPERMARKET]++;
    else if (place.types.indexOf(GAS_STATION) != -1) badgeCount[GAS_STATION]++;
    else if (place.types.indexOf(SHOPPING_MALL) != -1)
      badgeCount[SHOPPING_MALL]++;
  });

  return (
    <Box marginTop={2} style={{ height: '100%' }}>
      <Box
        display="flex"
        flexDirection="column"
        alignContent="space-between"
        style={{ height: '100%' }}
      >
        <Box flexGrow={1}>
          <FormControl component="fieldset" style={{ width: '100%' }}>
            <FormLabel className={classes.legend} component="legend">
              Choose a type of place
            </FormLabel>
            <FormGroup>
              <List>
                {Object.keys(placeTypes).map((key) => (
                  <ListItem key={key}>
                    <ListItemIcon>
                      <Badge
                        badgeContent={badgeCount[placeTypes[key].type]}
                        color="primary"
                      >
                        {getIcon(placeTypes[key].type)}
                      </Badge>
                    </ListItemIcon>
                    <ListItemText primary={placeTypes[key].label} />
                    <ListItemSecondaryAction>
                      <FormControlLabel
                        control={
                          <Switch
                            edge="end"
                            checked={placeTypes[key].isOn}
                            onChange={handleChange}
                            inputProps={{
                              'aria-labelledby': placeTypes[key].type,
                            }}
                            name={placeTypes[key].type}
                          />
                        }
                      />
                    </ListItemSecondaryAction>
                  </ListItem>
                ))}
              </List>
            </FormGroup>
            {/* <FormHelperText>Be careful</FormHelperText> */}
          </FormControl>
        </Box>
        <Box textAlign="center" padding={2}>
          <Grid
            component="div"
            container
            style={{ justifyContent: 'center', paddingBottom: '16px' }}
            alignItems="center"
            spacing={1}
          >
            <Grid item>
              <Tooltip title="Light theme">
                <Brightness5Icon />
              </Tooltip>
            </Grid>
            <Grid item>
              <Switch
                checked={theme === 'dark'}
                onChange={toggleTheme}
                name="theme"
              />
            </Grid>
            <Grid item>
              <Tooltip title="Dark theme">
                <Brightness3Icon />
              </Tooltip>
            </Grid>
          </Grid>
          <Box
            display="flex"
            alignItems="center"
            justifyContent="center"
            paddingBottom={2}
          >
            <GitHubIcon fontSize="small" />
            <Box paddingLeft={1}>
              <Link
                color="textPrimary"
                href="https://github.com/tushar2289"
                onClick={(event) => event.preventDefault()}
              >
                tushar2289
              </Link>
            </Box>
          </Box>
          <Typography variant="body1">
            Made with <FavoriteIcon fontSize="small" color="error" /> in India
          </Typography>
        </Box>
      </Box>
    </Box>
  );
};

export default Drawer;
