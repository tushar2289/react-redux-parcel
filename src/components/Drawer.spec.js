import React from 'react';
import { Provider } from 'react-redux';
import { createShallow } from '@material-ui/core/test-utils';
import Drawer from './Drawer';
import store from '../__mocks__/mockStore';

describe('Drawer component', () => {
  let shallow;
  beforeAll(() => {
    shallow = createShallow();
  });

  it('should render properly', () => {
    const render = shallow(
      <Provider store={store}>
        <Drawer />
      </Provider>
    );
    expect(render.html()).toMatchSnapshot();
  });
});
