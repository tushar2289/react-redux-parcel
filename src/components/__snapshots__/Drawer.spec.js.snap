// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component should render properly 1`] = `"<div class=\\"MuiBox-root MuiBox-root-2\\" style=\\"height:100%\\"><div class=\\"MuiBox-root MuiBox-root-3\\" style=\\"height:100%\\"><div class=\\"MuiBox-root MuiBox-root-4\\"><fieldset class=\\"MuiFormControl-root\\" style=\\"width:100%\\"><legend class=\\"MuiFormLabel-root makeStyles-legend-1\\">Choose a type of place</legend><div class=\\"MuiFormGroup-root\\"><ul class=\\"MuiList-root MuiList-padding\\"><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M11 9H9V2H7v7H5V2H3v7c0 2.12 1.66 3.84 3.75 3.97V22h2.5v-9.03C11.34 12.84 13 11.12 13 9V2h-2v7zm5-3v8h2.5v8H21V2c-2.76 0-5 2.24-5 4z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">12</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Restaurant</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-6 Mui-checked\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" checked=\\"\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"restaurant\\" aria-labelledby=\\"restaurant\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M4 10h3v7H4zM10.5 10h3v7h-3zM2 19h20v3H2zM17 10h3v7h-3zM12 1L2 6v2h20V6z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">1</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Bank</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"bank\\" aria-labelledby=\\"bank\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 3H5c-1.1 0-1.99.9-1.99 2L3 19c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-1 11h-4v4h-4v-4H6v-4h4V6h4v4h4v4z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">1</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Hospital</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"hospital\\" aria-labelledby=\\"hospital\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M11 17h2v-1h1c.55 0 1-.45 1-1v-3c0-.55-.45-1-1-1h-3v-1h4V8h-2V7h-2v1h-1c-.55 0-1 .45-1 1v3c0 .55.45 1 1 1h3v1H9v2h2v1zm9-13H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4V6h16v12z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">1</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">ATM</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"atm\\" aria-labelledby=\\"atm\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M22 16v-2l-8.5-5V3.5c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5V9L2 14v2l8.5-2.5V19L8 20.5V22l4-1 4 1v-1.5L13.5 19v-5.5L22 16z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">1</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Airport</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"airport\\" aria-labelledby=\\"airport\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 2c-4 0-8 .5-8 4v9.5C4 17.43 5.57 19 7.5 19L6 20.5v.5h2.23l2-2H14l2 2h2v-.5L16.5 19c1.93 0 3.5-1.57 3.5-3.5V6c0-3.5-3.58-4-8-4zM7.5 17c-.83 0-1.5-.67-1.5-1.5S6.67 14 7.5 14s1.5.67 1.5 1.5S8.33 17 7.5 17zm3.5-7H6V6h5v4zm2 0V6h5v4h-5zm3.5 7c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">1</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Train station</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"train_station\\" aria-labelledby=\\"train_station\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">1</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Supermarket</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"supermarket\\" aria-labelledby=\\"supermarket\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19.77 7.23l.01-.01-3.72-3.72L15 4.56l2.11 2.11c-.94.36-1.61 1.26-1.61 2.33 0 1.38 1.12 2.5 2.5 2.5.36 0 .69-.08 1-.21v7.21c0 .55-.45 1-1 1s-1-.45-1-1V14c0-1.1-.9-2-2-2h-1V5c0-1.1-.9-2-2-2H6c-1.1 0-2 .9-2 2v16h10v-7.5h1.5v5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V9c0-.69-.28-1.32-.73-1.77zM12 10H6V5h6v5zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">1</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Fuel pump</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"gas_station\\" aria-labelledby=\\"gas_station\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li><li class=\\"MuiListItem-container\\"><div class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction\\"><div class=\\"MuiListItemIcon-root\\"><span class=\\"MuiBadge-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 6h-2c0-2.76-2.24-5-5-5S7 3.24 7 6H5c-1.1 0-1.99.9-1.99 2L3 20c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-7-3c1.66 0 3 1.34 3 3H9c0-1.66 1.34-3 3-3zm0 10c-2.76 0-5-2.24-5-5h2c0 1.66 1.34 3 3 3s3-1.34 3-3h2c0 2.76-2.24 5-5 5z\\"></path></svg><span class=\\"MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary\\">1</span></span></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Mall</span></div></div><div class=\\"MuiListItemSecondaryAction-root\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root MuiSwitch-edgeEnd\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"shopping_mall\\" aria-labelledby=\\"shopping_mall\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\"></span></label></div></li></ul></div></fieldset></div><div class=\\"MuiBox-root MuiBox-root-9\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center\\" style=\\"justify-content:center;padding-bottom:16px\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"Light theme\\"><path d=\\"M20 15.31L23.31 12 20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6z\\"></path></svg></div><div class=\\"MuiGrid-root MuiGrid-item\\"><span class=\\"MuiSwitch-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-6 Mui-checked\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" checked=\\"\\" class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"theme\\"/><span class=\\"MuiSwitch-thumb\\"></span></span></span><span class=\\"MuiSwitch-track\\"></span></span></div><div class=\\"MuiGrid-root MuiGrid-item\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" title=\\"Dark theme\\"><path d=\\"M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.06 7 9.54 0 4.48-2.94 8.27-7 9.54.95.3 1.95.46 3 .46 5.52 0 10-4.48 10-10S14.52 2 9 2z\\"></path></svg></div></div><div class=\\"MuiBox-root MuiBox-root-10\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3\\"></path></svg><div class=\\"MuiBox-root MuiBox-root-11\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorTextPrimary\\" href=\\"https://github.com/tushar2289\\">tushar2289</a></div></div><p class=\\"MuiTypography-root MuiTypography-body1\\">Made with <svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z\\"></path></svg> in India</p></div></div></div>"`;
